### ISSUE TITLE
Retain X number of lambda functions on update call

### ISSUE THREAD

**Issue by mcrowson** [Miserlou/Zappa#1402]
## Context

AWS Lambda retains old copies of code. This makes it easier to rollback, but its also possible to reach the 75G cap of stored lambda functions if a CI pipeline is constantly updating lambda code.

We should add a key to the zappa_settings.json for `retained_versions` or something similarly named that takes an integer and when `zappa update` is called, we prune the older versions greater than the `retained_versions` value.

## Expected Behavior

Zappa cleans up old versions of lambda functions when the update command is called.

## Actual Behavior

No pruning of old versions happens.

**Comment by harryd31** [comment]
I completely agree, actually ran into that issue last week. Had to manually delete previous versions.

**Comment by drewstreib** [comment]
I do this in a very mundane way today, and manually prune wide ranges:

#!/bin/bash
for i in {201..300}; do
  aws lambda delete-function --function-name $1 --qualifier $i 
  echo "lambda $1 version $i was deleted"
done

**Comment by party98** [comment]
Hey, I'd like to take this up. Please help me as to where should i start.

**Comment by mcrowson** [comment]
I see it as a few changes:
1. Add an argument to the zappa_settings.json file (update readme to show that it is there, and its default)
2. Add an optional parameter to the [update_lambda_function](https://github.com/Miserlou/Zappa/blob/master/zappa/core.py#L1067) that takes the number of revisions to keep.
3. Get the revisions_to_keep setting from the stage config and pass it to all calls of the update_lambda_function from the [cli's update function](https://github.com/Miserlou/Zappa/blob/master/zappa/cli.py#L836)
4. After the [function gets updated](https://github.com/Miserlou/Zappa/blob/master/zappa/core.py#L1083), get the value from the settings then make a call to get the[ versions currently in lambda](http://boto3.readthedocs.io/en/latest/reference/services/lambda.html#Lambda.Client.list_versions_by_function) for this function.
5. Compare the number of functions desired and the number of functions in lambda and then make a call to [lambda to delete](http://boto3.readthedocs.io/en/latest/reference/services/lambda.html#Lambda.Client.delete_function) the oldest versions in excess of the desired amount.

That's how I'd approach it anyhow. I might be missing something, and if it possible to write a test for this please do that too.

**Comment by party98** [comment]
Thanks. On it

**Comment by party98** [comment]
@mcrowson Can you elaborate on what is stage config.

**Comment by dmateos** [comment]
I see this has been implemented, generally how long before stuff like this shows up in PIP?
